rest/src/main/java/fi/otavanopisto/pyramus/rest/StudentRESTService.java

```diff
@@ -2540,6 +2540,57 @@ public Response findStudentAddress(@PathParam("STUDENTID") Long studentId, @Path
   }
 
   @Path("/students/{STUDENTID:[0-9]*}/addresses/{ID:[0-9]*}")
+  @PUT
+  @RESTPermit(handling = Handling.INLINE)
+  // @RESTPermit (StudentPermissions.UPDATE_STUDENTADDRESS)
+  public Response updateStudentAddress(
+      @PathParam("STUDENTID") Long studentId,
+      @PathParam("ID") Long id,
+      fi.otavanopisto.pyramus.rest.model.Address body
+      ) {
+    Student student = studentController.findStudentById(studentId);
+    if (student == null) {
+      return Response.status(Status.NOT_FOUND).build();
+    }
+
+    if (student.getArchived()) {
+      return Response.status(Status.NOT_FOUND).build();
+    }
+
+    if (!restSecurity.hasPermission(
+          new String[] { StudentPermissions.UPDATE_STUDENTADDRESS },
+          student)
+        && !restSecurity.hasPermission(
+            new String[] { PersonPermissions.PERSON_OWNER },
```

Comment:
I'd say STUDENT_OWNER. Address getters seem happy with just PERSON_OWNER but address creation requires STUDENT_OWNER and so updating should, too.